### PR TITLE
[T7136]bad_connector_woocommerce:WooCommerce Webhook BAD-SIP.

### PR DESCRIPTION
**Pull request by bizzappdevteam**
- Generate the token to get the "Access Token".
- Follow the step which is mention in "Webhook Configuration" Tab.

### CODE REVIEW

File: bad_connector_woocommerce/models/woo_backend/common.py [hunk 1]
```diff
@@ -179,6 +180,44 @@ def _get_stock_field_id(self):
     woo_setting_id = fields.Many2one(comodel_name="woo.settings")
     stock_update = fields.Boolean(related="woo_setting_id.stock_update")
     recompute_qty_step = fields.Integer(string="Recompute Quantity Batch", default=1000)
+    access_token = fields.Char(string="Access Token", readonly=True)
```
Comment (bizzappdevteam) [inline bad_connector_woocommerce/models/woo_backend/common.py]: Don't we need test_access_token field to handle the test mode? your suggestion please Ruchir

File: bad_connector_woocommerce/tests/test_woo_backend.py [hunk 2]
```diff
@@ -109,3 +310,40 @@ def test_backend_cron(self):
         self.backend.cron_export_sale_order_status()
         self.backend.cron_update_stock_qty()
         self.backend.cron_import_product_templates()
+        self.backend.generate_token()
+
+    def test_product_create_webhook(self):
+        """Called webhook for Product"""
+        product_webhook_url = "/update_product/woo_webhook/{}".format(
+            self.backend.access_token
```
Comment (bizzappdevteam) [inline bad_connector_woocommerce/tests/test_woo_backend.py]: I think here, we have to pass the self.backend.test_acess_token, as the backend currently is in test mode, if it is, then do it all places.

File: bad_connector_woocommerce/tests/test_woo_backend.py [hunk 3]
```diff
@@ -28,6 +33,7 @@ def setUp(self):
                 "tracking_info": True,
                 "warehouse_id": warehouse.id,
                 "update_stock_inventory": True,
+                "access_token": "d4ea64d3-8f85-4955-be49-4aeb29151801",
```
Comment (bizzappdevteam) [inline bad_connector_woocommerce/tests/test_woo_backend.py:63]: Don't we have to add test_access_token as well, to cover the tests

File: bad_connector_woocommerce/controllers/main.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+from odoo.http import request
+
+_logger = logging.getLogger(__name__)
+
+
+class WooWebhook(http.Controller):
+    def _common_webhook_handler(self, access_token, model_name):
+        """Common handler for processing WooCommerce webhooks."""
+        job_options = {}
+        payload = json.loads(request.httprequest.data)
```
Comment (bizzappdev) [inline bad_connector_woocommerce/controllers/main.py]: do this only if valid access token

File: bad_connector_woocommerce/models/woo_backend/common.py [hunk 5]
```diff
@@ -179,6 +180,45 @@ def _get_stock_field_id(self):
...
+        <ol>
+            <li>Navigate to WooCommerce > Settings > Advanced > Webhooks.</li>
+            <li>Click "Add Webhook" and provide the following details:</li>
+        </ol>
+        <ul>
+            <li><strong>Name:</strong> Update Product</li>
+            <li><strong>Status:</strong> Active</li>
+            <li><strong>Topic:</strong> Product Update</li>
+            <li><strong>Delivery URL:</strong>
+            http://www.website.com:8080/update_product/woo_webhook/d4ea64d3-8f85-
```
Comment (bizzappdev) [inline bad_connector_woocommerce/models/woo_backend/common.py]: this url has to be dymanic calculated from the web based URL and token current value

File: bad_connector_woocommerce/tests/test_woo_backend.py (hunk 3 above)
Comment (bizzappdev) [reply bad_connector_woocommerce/tests/test_woo_backend.py:63]: you have skipped this comment

File: bad_connector_woocommerce/controllers/main.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+from odoo import http
+from odoo.http import request
+
+_logger = logging.getLogger(__name__)
+
+
+class WooWebhook(http.Controller):
+    def _common_webhook_handler(self, access_token, model_name):
+        """Common handler for processing WooCommerce webhooks."""
+        job_options = {}
```
Comment (bizzappdev) [inline bad_connector_woocommerce/controllers/main.py:15]: if not access token then also Forbidden even before searching

File: bad_connector_woocommerce/controllers/main.py [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        )
+        if not backend:
+            _logger.error(
+                "No WooCommerce backend found. Check your Access Token and try again"
+            )
+            raise Forbidden()
+        base_url = request.httprequest.base_url
+        parsed_url = urlparse(base_url)
+        base_url = f"{parsed_url.scheme}://{parsed_url.netloc}"
+        backend.write({"base_url": base_url})
```
Comment (bizzappdev) [inline bad_connector_woocommerce/controllers/main.py]: why this is needed?

Comment (bizzappdevteam) [reply bad_connector_woocommerce/controllers/main.py]: We don't need this. I will remove it and take the web.base.url

File: bad_connector_woocommerce/controllers/main.py [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+        if not backend:
+            _logger.error(
+                "No WooCommerce backend found. Check your Access Token and try again"
+            )
+            raise Forbidden()
+        payload = json.loads(request.httprequest.data)
+        if model_name == "woo.sale.order":
+            status = backend.woo_sale_status_ids.mapped("code")
+            if status and payload.get("status") not in status:
+                return True
```
Comment (bizzappdev) [inline bad_connector_woocommerce/controllers/main.py:53]: add logger for skipping.

**Review by bizzappdev — CHANGES_REQUESTED** [review]

File: bad_connector_woocommerce/controllers/main.py [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+                "No WooCommerce backend found. Check your Access Token and try again"
+            )
+            raise Forbidden()
+        payload = json.loads(request.httprequest.data)
+        payload_status = payload.get("status")
+        if model_name == "woo.sale.order":
+            status = backend.woo_sale_status_ids.mapped("code")
+            if status and payload_status not in status:
+                _logger.info(
+                    "Skipping sale order import due to status: %s",
```
Comment (bizzappdevteam) [inline bad_connector_woocommerce/controllers/main.py]:
```suggestion
                    "Skipping Sale Order Import due to status %s is not found",
```

File: bad_connector_woocommerce/controllers/main.py [hunk 10]
```diff
@@ -46,7 +46,7 @@ def _common_webhook_handler(self, access_token, model_name):
             status = backend.woo_sale_status_ids.mapped("code")
             if status and payload_status not in status:
                 _logger.info(
-                    "Skipping sale order import due to status: %s",
+                    "Skipping sale order import due to status %s is not found",
```
Comment (bizzappdev) [inline bad_connector_woocommerce/controllers/main.py]: Skipping sale order import due to status %s is not configured for import

**Comment by bizzappdevteam** [comment]
Closing this PR due to this functionality is there in https://github.com/BizzAppDev-Systems/connector-woocommerce/pull/71

**Pull request closed without merging**